src/rules/no-invalid-import-placement.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * @fileoverview Rule to enforce correct placement of `@import` rules in CSS.
+ * @author thecalamiity
+ */
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @import { CSSRuleDefinition } from "../types.js"
+ * @typedef {"invalidImportPlacement"} NoInvalidImportPlacementMessageIds
+ * @typedef {CSSRuleDefinition<{ RuleOptions: [], MessageIds: NoInvalidImportPlacementMessageIds }>} NoInvalidImportPlacementRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+const IGNORED_AT_RULES = new Set(["charset", "layer"]);
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {NoInvalidImportPlacementRuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			description: "Disallow invalid placement of @import rules",
+			recommended: true,
+			url: "https://github.com/eslint/css/blob/main/docs/rules/no-invalid-import-placement.md",
+		},
+
+		messages: {
+			invalidImportPlacement:
+				"@import must be placed before all other rules, except @charset and @layer.",
+		},
+	},
+
+	create(context) {
+		let hasSeenNonImportRule = false;
+
+		return {
+			Atrule(node) {
+				const name = node.name.toLowerCase();
+
+				if (IGNORED_AT_RULES.has(name)) {
+					return;
+				}
```

Comment:
I think it won't report @import if it is placed before @charset and @layer, which is also an invalid placement. Can you fix that?